clang/lib/Analysis/CFG.cpp

```diff
@@ -443,21 +443,44 @@ class reverse_children {
 
 } // namespace
 
-reverse_children::reverse_children(Stmt *S) {
-  if (CallExpr *CE = dyn_cast<CallExpr>(S)) {
-    children = CE->getRawSubExprs();
+reverse_children::reverse_children(Stmt *S, ASTContext &Ctx) {
+  switch (S->getStmtClass()) {
+  case Stmt::CallExprClass: {
```

Comment:
Is this equivalent what we had before? What about the different subclasses of `CallExpr`?

Reply:
In the past it was implicit in the `dyn_cast<CallExpr>`. So it's the same as before.

Reply:
I think `dyn_cast` would succeed for `CXXMemberCallExpr` and similar subclasses because `classof` is implemented  like:
```
  static bool classof(const Stmt *T) {
    return T->getStmtClass() >= firstCallExprConstant &&
           T->getStmtClass() <= lastCallExprConstant;
  }
  ```
On the other hand, I'd expect the stmt class to be `CallExprClass` only when it is not a derived type.